### DISCUSSION TITLE
Settings window retains Dark theme when other theme is selected

### DISCUSSION THREAD

**Discussion by ktnjared** [Chatterino/chatterino2#4823] [Ideas] [answered]
### Checklist

- [X] I'm reporting a problem with Chatterino
- [X] I've verified that I'm running **the most** recent nightly build or stable release
- [X] I've looked for my problem on the [wiki](https://wiki.chatterino.com/Help/)
- [X] I've searched the [issues and pull requests](https://github.com/Chatterino/chatterino2/issues?q=) for similar looking reports

### Describe your issue

With any theme other than the default "Dark" theme is selected, the changes are not reflected within the Settings window. The main/chat window immediately reflects the change in theme. The issue continues after quitting and restarting after choosing the theme.

This is a UX usability issue for users who have difficulties reading light text on dark backgrounds, particularly those with dyslexia or astigmatism.

I verified that the value is truly set in `~/Library/Application\ Support/chatterino/Settings/settings.json`:
```json
"misc": {
    "currentVersion": "2.4.5",
    "lockNotebookLayout": false,
    "beta": true,
    "restartOnCrash": true
},
"appearance": {
    "theme": {
        "name": "Light"
    },
    "currentFontFamily": "SF Mono Square",
    "currentFontSize": 10,
    "showTabCloseButton": false,
    "hidePreferencesButton": true,
    "hideUserButton": true,
    "messages": {
        "separateMessages": true,
        "alternateMessageBackground": true,
        "hideModerated": true,
        "showTimestamps": true,
        "timestampFormat": "hh:mm"
    },
    "splitheader": {
        "showUptime": true,
        "showGame": true,
        "showTitle": true
    },
    "uiScale2": 1.200000047683716,
    "tabDirection": 2
},
```

Tested on
- macOS Ventura (13.5.2 (22G91))
	- Chatterino 2.4.5 (commit 05a72d33) built with Qt 6.5.0 on on macOS 13.5.2, kernel: 22.6.0
	- Chatterino 2.4.5 (commit d752ce86) built with Qt 5.15.2 on on macOS 13.5.2, kernel: 22.6.0
	- Chatterino 2.4.5 (commit d752ce86) built with Qt 6.5.0 on on macOS 13.5.2, kernel: 22.6.0
- macOS Sonoma (14.0 (23A339))
	- Chatterino 2.4.5 (commit 05a72d33) built with Qt 6.5.0 on on macOS 14.0, kernel: 23.0.0
	- Chatterino 2.4.5 (commit d752ce86) built with Qt 5.15.2 on on macOS 14.0, kernel: 23.0.0
	- Chatterino 2.4.5 (commit d752ce86) built with Qt 6.5.0 on on macOS 14.0, kernel: 23.0.0

### Screenshots

<img width="939" alt="Screenshot 2023-09-16 at 22 19 57" src="https://github.com/Chatterino/chatterino2/assets/5382149/68cf674b-507e-4176-a630-250ee6fc0dd1">

### OS and Chatterino Version

Chatterino 2.4.5 (commit d752ce86) built with Qt 6.5.0 on on macOS 14.0, kernel: 23.0.0

**Comment by Felanbird** [comment]
This isn't a bug there just isn't code for this window to respect the `Theme` setting, this was the case with many other windows until neri started implementing them #4423 #4671 #4817

**Comment by pajlada** [comment]
There is some concern about theming the settings window since that's where you configure your theme. If you configure your theme to be unreadable (since it's not something we check for), you would be forced to modify the settings file yourself or just delete it and start again or just deleting your theme.

Would it be ok if the settings dialog only had two built-in themes: dark mode (as it is now) and light mode? so no white or black mode, and no custom themes

**Reply by Mm2PL** [reply]
We could just add a safe mode (kinda like RuneLite does) which doesn't load any external themes or plugins. I believe this is a non-concern.

**Reply by pajlada** [reply]
I wasn't saying it's not technically feasible, but I do believe it's not worth the effort of implementation of maintainability.
I am still OK with the settings window respecting your chosen theme's Light or Dark mode value & using that.

**Comment by kwijybo** [comment]
It would be really helpful if this were implemented please, I need light to see for accessibility. Just having Light and Dark and nothing else for the settings window would definitely be good enough.

**Reply by pajlada** [reply]
I am OK with this being implemented

**Reply by pajlada** [reply, marked as answer]
I've moved this segment into an issue https://github.com/Chatterino/chatterino2/issues/7117 , since that's what I'm fine with being implemented.
Thanks for bringing your accessibility issue up.
